wordle-web/pages/game.vue

```diff
@@ -1,105 +1,215 @@
+    >
+
+    <v-row justify="center">
+      <game-board :wordleGame="wordleGame" />
+    </v-row>
+    <v-row justify="center">
+      <keyboard :wordleGame="wordleGame" />
+    </v-row>
   </v-container>
 </template>
 
 <script lang="ts">
-import { Component, Vue } from 'vue-property-decorator'
+import { Component, Vue, Watch } from 'vue-property-decorator'
 import { WordsService } from '~/scripts/wordsService'
 import { GameState, WordleGame } from '~/scripts/wordleGame'
 import KeyBoard from '@/components/keyboard.vue'
 import GameBoard from '@/components/game-board.vue'
 import { Word } from '~/scripts/word'
+import { Stopwatch } from 'ts-stopwatch'
 
 @Component({ components: { KeyBoard, GameBoard } })
 export default class Game extends Vue {
   word: string = WordsService.getRandomWord()
   wordleGame = new WordleGame(this.word)
-
-  isLoaded: boolean = false
+  username: string = this.evaluateName()
+  dialog: boolean = false
+  submittedSuccess: boolean = false
+  gameTimeSeconds: number = 0
+  stopwatch = new Stopwatch()
+  rules = {
+    required: (value: any) => !!value || 'Required.',
+    counter: (value: string | any[]) =>
+      value.length <= 25 || 'Max 25 characters',
+  }
 
   mounted() {
-    setTimeout(() => {
-      this.isLoaded = true
-    }, 5000)
+    this.startTime()
+  }
+
+  startTime() {
+    this.stopwatch.start()
+  }
+
+  stopTime() {
+    this.stopwatch.stop()
+  }
+
+  getTime() {
+    return this.stopwatch.getTime()
+  }
+
+  resetTime() {
+    this.stopwatch.reset()
+  }
+
+  getTimeSeconds() {
+    return this.getTime() / 1000
   }
 
   resetGame() {
+    this.submittedSuccess = false
     this.word = WordsService.getRandomWord()
     this.wordleGame = new WordleGame(this.word)
+    this.resetTime()
+    this.startTime()
+  }
+
+  submitScore() {
+    //guard statement, only submit score once
+    if (this.submittedSuccess) return
+
+    if (!this.hasUniqueLocalName()) {
+      this.toggleDialog()
+      this.waitUntilDialogClosed()
+    } else {
+      this.$axios.post('/api/PlayersLeaderBoard', {
+        name: this.username,
+        attempts: this.wordleGame.currentGuess,
+        seconds: this.getTimeSeconds().toFixed(0), //removes after numbers decimal
+      })
+      this.submittedSuccess = true
+    }
+  }
+
+  waitUntilDialogClosed() {
+    if (this.dialog) {
+      //we want it to be false
+      setTimeout(this.waitUntilDialogClosed, 1000) //wait 50 millisecnds then recheck
+      return
+    }
+    this.$axios.post('/api/PlayersLeaderBoard', {
+      name: this.username,
+      attempts: this.wordleGame.currentGuess,
+      seconds: this.getTimeSeconds().toFixed(0), //removes after numbers decimal
+    })
   }
 
   get gameResult() {
     if (this.wordleGame.state === GameState.Won) {
-      return { type: 'success', text: 'You won! :^)' }
+      this.stopTime()
+      return {
+        type: 'success',
+        text: 'You won the game in ' + this.getTimeSeconds() + ' seconds!',
+      }
```

Comment:
It would be cool to have a timer on the game page to display the time as you play. This will also let new players know they are being timed and to try to go as fast as possible or for returning players to monitor their progress on the word.